src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -904,7 +904,7 @@ export class List<T> implements ISpliceable<T>, IDisposable {
 
 		this.onDidFocus = mapEvent(domEvent(this.view.domNode, 'focus', true), () => null);
 		this.onDidBlur = mapEvent(domEvent(this.view.domNode, 'blur', true), () => null);
-
+		this.onDidFocus(() => this.focusFirstItemIfNothingFocused(), this, this.disposables);
```

Comment:
I really don't think this is the best behaviour for the list. The default behaviour of DOM focus should be just that, not focusing an element.